test/ListProxyTest.js

```diff
@@ -0,0 +1,266 @@
+var expect = require("chai").expect;
+var HazelcastClient = require("../.").Client;
+var Controller = require('./RC');
+var Util = require('./Util');
+
+describe("List Proxy", function () {
+
+    var cluster;
+    var client;
+    var listInstance;
+
+    before(function () {
+        this.timeout(10000);
+        return Controller.createCluster().then(function (response) {
+            cluster = response;
+            return Controller.startMember(cluster.id);
+        }).then(function () {
+            return HazelcastClient.newHazelcastClient().then(function (hazelcastClient) {
+                client = hazelcastClient;
+            });
+        });
+    });
+
+    beforeEach(function () {
+        listInstance = client.getList('test')
+    });
+
+    afterEach(function () {
+        return listInstance.destroy();
+    });
+
+    after(function () {
+        client.shutdown();
+        return Controller.shutdownCluster(cluster.id);
+    });
+
+
+    it("appends one item", function () {
+        return listInstance.add(1).then(function () {
+            return listInstance.size().then(function (size) {
+                expect(size).to.equal(1);
+            });
+        })
+    });
+
+    it("inserts one item at index", function () {
+        return listInstance.addAll([1, 2, 3]).then(function () {
+            return listInstance.addAt(1, 5).then(function (size) {
+                return listInstance.toArray();
+            }).then(function (all) {
+                expect(all).to.deep.equal([1, 5, 2, 3]);
+            });
+        })
+    });
+
+    it("clears", function () {
+        return listInstance.addAll([1, 2, 3]).then(function () {
+            return listInstance.clear();
+        }).then(function () {
+            return listInstance.size().then(function (size) {
+                expect(size).to.equal(0);
+            });
+        });
+    });
+
+
+    it("inserts all elements of array at index", function () {
+        return listInstance.addAll([1, 2, 3]).then(function () {
+            return listInstance.addAllAt(1, [5, 6]);
+        }).then(function () {
+            return listInstance.toArray();
+        }).then(function (all) {
+            expect(all).to.deep.equal([1, 5, 6, 2, 3])
+        });
+    });
+
+
+    it("gets item at index", function () {
+        var input = [1, 2, 3];
+        return listInstance.addAll(input).then(function () {
+            return listInstance.get(1).then(function (result) {
+                expect(result).to.equal(2);
+            });
+        });
+    });
+
+    it("removes item at index", function () {
```

Comment:
This same test can be and should be written as
```
    it("removes item at index", function () {
        var input = [1, 2, 3];
        return listInstance.addAll(input).then(function () {
            return listInstance.removeAt(1);
        }).then(function (removed) {
            expect(removed).to.equal(2);
            return listInstance.toArray();
        }).then(function(arr) {
            expect(arr).to.deep.equal([1, 3]);
        });
    });
```
It is much easier to read. Also this is why we use promises for. Current test misses the point in using promises.